getnacelle/nacelle-js
# refactor: remove `@urql/exchange-persisted` patch

## Why are these changes introduced?

Addresses [ENG-9551](https://nacelle.atlassian.net/browse/ENG-9551) 

> Upgrade `@urql/exchange-persisted` and remove patch



Storefront SDK `2.0.0` launched with a patched version of `@urql/exchange-persisted@3.0.1` (see https://github.com/getnacelle/nacelle-js/pull/379). Now that the underlying issue with `@urql/exchange-persisted` has been resolved by the latest version (`4.0.0`), we can ditch the patch. This should result in faster `npm install`s for projects that use the Storefront SDK, because there is no need to download and run `patch-package`.

## What is this pull request doing?



This PR upgrades `@urql/exchange-persisted` to take advantage of the changes from https://github.com/urql-graphql/urql/pull/3192.

This PR also updates the Next.js Reference Store to use the latest version of the Storefront SDK.

## How to Test



1. Check out the `ENG-9551/remove-urql-exchange-persisted-patch` branch
2. `npm run bootstrap` from the monorepo root
3. Navigate to `packages/storefront-sdk`
4. `npm run build` - the package should build without errors
5. `npm run coverage` - all tests should pass with 100% coverage
6. Navigate to `packages/storefront-sdk-plugins/commerce-queries`
7. `npm run build` - the package should build without errors
8. `npm run coverage` - all tests should pass with 100% coverage
9. Navigate to `reference-stores/next`
10. `npm run build` - the project should build without errors. Previously, the project would error if using the older version of `@urql/exchange-persisted` without a patch.

## Checklist



[ENG-9551]: https://nacelle.atlassian.net/browse/ENG-9551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
